WindowsUniversal/WindowsUniversal/MainPage.xaml.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Microsoft.AspNetCore.SignalR.Client;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+
+namespace WindowsUniversal
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class MainPage : Page
+    {
+        private HubConnection _connection;
+
+        public MainPage()
+        {
+            this.InitializeComponent();
+        }
+
+        private async void ConnectButton_Click(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                _connection = new HubConnectionBuilder()
+                    .WithUrl(UrlTextBox.Text)
+                    .ConfigureLogging(logging =>
+                    {
+                        logging.AddDebug();
+                        logging.SetMinimumLevel(LogLevel.Trace);
+                    })
+                    .Build();
+
+                var syncContext = SynchronizationContext.Current;
+                _connection.On<string>("Send", s =>
+                {
+                    syncContext.Post((state) => AppendMessage((string)state), s);
```

Comment:
Is there no Dispatcher.Invoke on UWP?

Reply:
There is `CoreDispatcher` available on `Page`. Using the dispatcher instead of the sync context is more idiomatic UWP.

Reply:
:+1:

Reply:
![https://i.gifer.com/BkF.gif](https://i.gifer.com/BkF.gif)